scripts/globals/weaponskills.lua

```diff
@@ -1153,6 +1157,30 @@ end
 
 -- Given the raw ratio value (atk/def) and levels, returns the cRatio (min then max)
 xi.weaponskills.cMeleeRatio = function(attacker, defender, params, ignoredDef, tp)
+    local weaponType = attacker:getWeaponSkillType(xi.slot.MAIN)
+    local pDifWeaponCapTable =
```

Comment:
this table shouldn't be copied -- you can change physical_utilities like so:

```diff
diff --git a/scripts/globals/combat/physical_utilities.lua b/scripts/globals/combat/physical_utilities.lua
index 8ad5631918..612c4aa990 100644
--- a/scripts/globals/combat/physical_utilities.lua
+++ b/scripts/globals/combat/physical_utilities.lua
@@ -46,7 +46,7 @@ local wsElementalProperties =
 }
 
 -- Table with pDIF caps per weapon/skill type.
-local pDifWeaponCapTable =
+xi.combat.physical.pDifWeaponCapTable =
 {
     -- [Skill/weapon type used] = {pre-randomizer_pDIF_cap}, Values from: https://www.bg-wiki.com/ffxi/PDIF
     [xi.skill.NONE            ] = { 3    }, -- We will use this for mobs.
@@ -384,7 +384,7 @@ xi.combat.physical.calculateMeleePDIF = function(actor, target, weaponType, wsAt
     local wRatio       = cRatio + (isCritical and 1.0 or 0)
     local pDifUpperCap = 0
     local pDifLowerCap = 0
-    local pDifFinalCap = pDifWeaponCapTable[weaponType][1] + (isCritical and 1.0 or 0) -- TODO: Add 'Damage Limit +' Trait here.
+    local pDifFinalCap = xi.combat.physical.pDifWeaponCapTable[weaponType][1] + (isCritical and 1.0 or 0) -- TODO: Add 'Damage Limit +' Trait here.
 
     -- pDIF upper cap.
     if wRatio < 0.5 then
@@ -516,7 +516,7 @@ xi.combat.physical.calculateRangedPDIF = function(actor, target, weaponType, wsA
     ----------------------------------------
     -- Step 4: Apply weapon type caps.
     ----------------------------------------
-    local pDifFinalCap = pDifWeaponCapTable[weaponType][1] -- TODO: Add 'Damage Limit +' Trait here.
+    local pDifFinalCap = xi.combat.physical.pDifWeaponCapTable[weaponType][1] -- TODO: Add 'Damage Limit +' Trait here.
 
     pDif = utils.clamp(pDif, 0, pDifFinalCap)
```

 then directly reference `xi.combat.physical.pDifWeaponCapTable`

Reply:
Ah, thanks for the help!